experimental/dds/tree2/src/test/shared-tree-core/branch.spec.ts

```diff
@@ -439,6 +435,59 @@ describe("Branches", () => {
 		assert.equal(branch.getHead().revision, nullRevisionTag);
 	});
 
+	describe("all nested forks and transactions are disposed and aborted when transaction is", () => {
+		const setUpNestedForks = (rootBranch: DefaultBranch) => {
+			change(rootBranch);
+			rootBranch.startTransaction();
+			const fork1 = rootBranch.fork();
```

Comment:
It would be nice to see the tests confirm that if the transaction is committed/aborted a second time, then fork1 is also disposed. These tests test "top-level/outer" transactions but not nested transactions.